client/blocks/comment-detail/comment-detail-header.jsx

```diff
@@ -82,7 +82,9 @@ export const CommentDetailHeader = ( {
 						</span>
 					</div>
 					<div className="comment-detail__author-info-element">
-						{ translate( 'on %(postTitle)s', { args: { postTitle } } ) }
+						{ translate( 'on %(postTitle)s', { args: {
+							postTitle: postTitle ? decodeEntities( postTitle ) : translate( 'Untitled' ),
```

Comment:
Might be worth noting that the current data structure doesn't include the post excerpt (but pretty much just the title).
If I'm not mistaken, we would need to introduce a new query component for posts just for those that don't have a title. So it likely makes more sense to do so backend side.